### ISSUE TITLE
Markdown2Json has some failures when transforming data

### ISSUE THREAD

**Issue by omiras** [IronHack-2024/CodeQuestAPI#58]
@kevmhughes pointed out that some of the questions in the MongoDB database are malformed.

![image](https://github.com/user-attachments/assets/dd89ac52-d6bc-4df2-993b-c46911517999)

In the example, the _answer_ fields are empty. However, we may encounter other types of errors.

1. We need to check the markdown2json script.  
2. We need to identify when a question is transformed unsuccessfully from Markdown to JSON and trigger a warning.  
3. No malformed questions should be inserted into the MongoDB database.  

The consequence of having malformed questions in the MongoDB database is that we cannot safely offer the `/api/questions/v1/random` endpoint to REST API consumers.

## Acceptance Criteria

- [ ] No malformed questions are inserted into the MongoDB database.  
- [ ] The markdown2json script must trigger a warning when it fails to transform a single question successfully.

**Comment by omiras** [comment]
@kevmhughes I have never come across a bad question in [https://codequestapi.onrender.com/daily-question](https://codequestapi.onrender.com/daily-question). Could it be possible that only questions with the **codeExamples** field filled are prone to being malformed?

**Comment by kevmhughes** [comment]
@omiras Yes, the codeExamples property is the problem here. However, it makes for very interesting questions, too. So far, I have seen 5 “types” of question that need to be processed differently to render them in the UI successfully:
type 1: 1 question, 0 code examples, 4 answers
type 2: 1 question, >1 and <4 code examples, 4 answers
type 3: 1 question, 4 code examples, 4 answers (A, B, C, D)
type 4: 1 question, 4 code examples, 4 answers (BUT only “:”)
type 5: 1 question, 5 code examples, 4 answers (A, B. C. D)

**Comment by kevmhughes** [comment]
@omiras After further investigation, I have seen that in fact there are a lot more than 5 “types” of question, including some with 8 possible answers, and even one “rogue” question that had 53 possible answers! LOL

To maintain coherence between the results in the responses given from both endpoints (https://codequestapi.onrender.com/api/v1/questions/ai, https://codequestapi.onrender.com/api/v1/questions/random), and after analysing the UI and UX when physically rendering 100s of questions in an app that consumes these endpoints, I strongly recommend that we only use two formats of question:

### Format 1:
questions with four possible answers and zero code examples:

<img width="574" alt="Screenshot 2024-12-10 at 04 54 30" src="https://github.com/user-attachments/assets/5d0be417-635b-4020-9767-46d5a4ec3c11">

### Format 2:
questions with four possible answers and one code example:

<img width="575" alt="Screenshot 2024-12-10 at 02 34 31" src="https://github.com/user-attachments/assets/55a81bd6-4d9b-4227-910b-f7200f19d068">

### How many Format 1 questions exist in our database?
The search below counted all the questions with 4 answers and 0 code examples in the `codequestapi-production` database in MongoDB and gave a result of 1217.
```
use('codequestapi-production');
db.getCollection('questions').countDocuments({
    "answerOptions": { $size: 4 },
    "codeExamples": { $size: 0 },
    "question": { $exists: true }
  });
```
### How many Format 2 questions exist in our database?
The search below counted all the questions with 4 answers and 1 code example in the `codequestapi-production` database in MongoDB and gave a result of 3449.
```
use('codequestapi-production');
db.getCollection('questions').countDocuments({
    "answerOptions": { $size: 4 },
    "codeExamples": { $size: 1 },
    "question": { $exists: true }
  });
```
### A solution for an MVP that can be offered to API consumers 
In conclusion, I believe the best way to offer an MVP to consumers is to simply delete (or exclude) all of the documents that do not correspond with the two formats mentioned previously from the `questions` collection in the `codequestapi-production` database in MongoDB. This solution does not address the `markdown2json` script issue, but it does mean that consumers can consume the API without further delay.
